shell/platform/android/android_surfacetransaction.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_ANDROID_ANDROID_SURFACETRANSACTION_H_
+#define FLUTTER_SHELL_PLATFORM_ANDROID_ANDROID_SURFACETRANSACTION_H_
+
+#include "flutter/fml/macros.h"
+
+#include <cstdint>
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// The Android SurfaceTransaction API is used by `AndroidSurface` to
+/// synchronize frame presentation times. It's only available on API 29+.
```

Comment:
How is this used to sychronize frame presentation times?

Reply:
It will turn all the changes applied during the transaction into one atomic update, that we can tag with the Vsync-ID.

Reply:
which changes?

Reply:
Any SurfaceControl which is updated in that time. It works the same way as "[CATransaction begin]" / "[CATransaction commit]".